curl-helper.c

```diff
@@ -3326,6 +3326,17 @@ SETOPT_LONG( BUFFERSIZE)
   SETOPT_STRING( AWS_SIGV4)
 #endif
 
+#if HAVE_DECL_CURLOPT_TCP_KEEPALIVE
+  SETOPT_BOOL( TCP_KEEPALIVE)
+#endif
+
+#if HAVE_DECL_CURLOPT_TCP_KEEPIDLE
+  SETOPT_INT( TCP_KEEPIDLE)
```

Comment:
did you try compiling? there is no SETOPT_INT
and then next question - was it tested?

Reply:
once i removed the conditional add `#if HAVE_DECL_CURLOPT_TCP_KEEPIDLE`, the compilation failed but with it the compile actually passed for me--might be some macro magic i'm not too sure.

i did a small test with a python server with the setup and i do see extra TCP requests dumped (when there wasn't any without the setup):
```
13:56:31.573793 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [S], seq 2756318970, win 65535, options [mss 65495,sackOK,TS val 3733098363 ecr 0,nop,wscale 14], length 0
E..<<i@.@..Q.............J...........0.........
...{........
13:56:31.573845 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 663868098, win 4, options [nop,nop,TS val 3733098363 ecr 3733098363], length 0
E..4<j@.@..X.............J..'........(.....
...{...{
13:56:31.573921 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [P.], seq 0:117, ack 1, win 4, options [nop,nop,TS val 3733098363 ecr 3733098363], length 117
E...<k@.@................J..'..............
...{...{GET / HTTP/1.1
Host: 127.0.0.1:3232
User-Agent: Curl_lwt
Accept: */*
Accept-Encoding: deflate, gzip, br, zstd


13:56:32.576286 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 1, win 4, options [nop,nop,TS val 3733099365 ecr 3733098363], length 0
E..4<l@.@..V.............J.o'........(.....
...e...{
13:56:35.616246 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 1, win 4, options [nop,nop,TS val 3733102405 ecr 3733099366], length 0
E..4<m@.@..U.............J.o'........(.....
...E...f
13:56:36.310771 IP ip6-localhost.40152 > ip6-localhost.3232: Flags [R], seq 3376754125, win 0, length 0
E..(..@.@.<..............E-.....P....S..
13:56:38.688243 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [.], ack 1, win 4, options [nop,nop,TS val 3733105477 ecr 3733102405], length 0
E..4<n@.@..T.............J.o'........(.....
...E...E
13:56:41.579093 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [F.], seq 117, ack 1, win 4, options [nop,nop,TS val 3733108368 ecr 3733105477], length 0
E..4<o@.@..S.............J.p'........(.....
.......E
13:56:46.321807 IP ip6-localhost.40154 > ip6-localhost.3232: Flags [R], seq 2756319089, win 0, length 0
E..(..@.@.<..............J.q....P...L...
```